docs/content/reference/resource.md

```diff
@@ -325,6 +325,139 @@ Example:
+
+Sweeper generation is enabled by default, except in the following conditions which require customization here:
+- Resources with custom deletion code
+- Resources with parent-child relationships (unless the parent relationship is configured)
+- Resources with complex URL parameters that aren't simple region/project parameters
+
+Simply defining the sweeper block in a resource will override these exclusions and enable sweeper generation for that resource, even with minimal configuration.
+
+### `exclude_sweeper`
+
+If set to `true`, no sweeper will be generated for this resource. This is useful for resources that cannot or should not be automatically cleaned up.
+
+Default: `false`
+
+Example:
+
+```yaml
+exclude_sweeper: true
+```
+
+### `sweeper`
+
+Configures how test resources are swept (cleaned up) after tests. The sweeper system helps ensure resources created during tests are properly removed, even when tests fail unexpectedly. All fields within the `sweeper` block are optional, with reasonable defaults provided when not specified.
+
+- `identifier_field`: Specifies which field in the resource object should be used to identify resources for deletion. If not specified, defaults to "name" if present in the resource, otherwise falls back to "id".
+
+- `prefixes`: Specifies name prefixes that identify resources eligible for sweeping. Resources whose names start with any of these prefixes will be deleted. By default, resources with the `tf-test-` prefix are automatically eligible for sweeping even if no prefixes are specified.
+
+- `url_substitutions`: Allows customizing URL parameters when listing resources. Each map entry represents a set of key-value pairs to substitute in the URL template. This is commonly used to specify regions to sweep in. If not specified, the sweeper will only run in the default region (us-central1) and zone (us-central1-a).
+
+- `dependencies`: Lists other resource types that must be swept before this one. This ensures proper cleanup order for resources with dependencies. If not specified, no dependencies are assumed.
+
+- `parent`: Configures sweeping for resources that depend on parent resources (like a nodepool that belongs to a cluster).
+
+  Required fields:
+  - `resource_type`: The type of the parent resource (e.g., "google_container_cluster")
+  - `child_field`: The field in your resource that references the parent (e.g., "cluster")
+  - At least one of `parent_field` or `template` is required
+
+  Options for getting parent reference:
+  - `parent_field`: The field from parent to use (typically "name" or "id")
```

Comment:
It would be worth clarifying here & for child_field whether this should be Terraform field name or API field name. Based on the examples it looks like they're likely API field name.

Reply:
will resolve with parent rework